libutils/string_lib.c

```diff
@@ -601,6 +602,120 @@ long StringToLongExitOnError(const char *str)
     return result;
 }
 
+/**
+ * @brief Converts a string of numerals in base 10 to a unsigned long 
+ *        integer.
+ *
+ * Result is stored in *value_out, return value should be checked.
+ * On error *value_out is unmodified and an error code is returned.
+ * Leading spaces in input string are skipped.
+ * String numeral must be terminated by NULL byte or space (isspace).
+ *
+ * @see StringToUlongExitOnError()
+ * @see StringToUlongDefaultOnError()
+ * @param[in] str String with numerals to convert, cannot be NULL
+ * @param[out] value_out Where to store result on success, cannot be NULL
+ * @return 0 on success, error code otherwise (see source code)
+ */
+int StringToUlong(const char *str, unsigned long *value_out)
+{
+    nt_static_assert(ERANGE != 0);
+    assert(str != NULL);
+    assert(value_out != NULL);
+
+    char *endptr = NULL;
+    unsigned long val;
+
+    errno = 0;
+    val = strtoul(str, &endptr, 10);
+
+    if ((errno == ERANGE && val == ULONG_MAX))
+    {
+        return ERANGE; // Overflow
+    }
+
+    // Negative numbers should cause underflow
+    if (val > 0) // -0 is ok
```

Comment:
```suggestion
    if (val > 0) // "-0" is ok
```

Nitpicking, but I suggest making it clear that you are talking about the string `"-0"` here.

Reply:
Agreed